server/functional-test/src/test/java/org/zanata/feature/administration/ManageSearchTest.kt

```diff
@@ -18,89 +18,93 @@
  * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA, or see the FSF
  * site: http://www.fsf.org.
  */
-package org.zanata.feature.administration;
+package org.zanata.feature.administration
 
-import org.junit.Before;
-import org.junit.Ignore;
-import org.junit.Test;
-import org.junit.experimental.categories.Category;
-import org.zanata.feature.Trace;
-import org.zanata.feature.testharness.TestPlan.DetailedTest;
-import org.zanata.feature.testharness.ZanataTestCase;
-import org.zanata.page.administration.ManageSearchPage;
-import org.zanata.page.dashboard.DashboardBasePage;
-import org.zanata.workflow.BasicWorkFlow;
-import org.zanata.workflow.LoginWorkFlow;
+import org.junit.Before
+import org.junit.Ignore
+import org.junit.Test
+import org.junit.experimental.categories.Category
+import org.zanata.feature.Trace
+import org.zanata.feature.testharness.TestPlan.DetailedTest
+import org.zanata.feature.testharness.ZanataTestCase
+import org.zanata.page.dashboard.DashboardBasePage
+import org.zanata.workflow.BasicWorkFlow
+import org.zanata.workflow.LoginWorkFlow
 
-import static org.assertj.core.api.Assertions.assertThat;
+import org.assertj.core.api.Assertions.assertThat
 
-@Category(DetailedTest.class)
-public class ManageSearchTest extends ZanataTestCase {
+/**
+ * @author Damian Jansen [djansen@redhat.com](mailto:djansen@redhat.com)
+ */
+@Category(DetailedTest::class)
+class ManageSearchTest : ZanataTestCase() {
 
-    private DashboardBasePage dashboardPage;
+    private var dashboardPage: DashboardBasePage? = null
 
     @Before
-    public void before() {
-        assertThat(new LoginWorkFlow().signIn("admin", "admin").loggedInAs())
-                .as("Admin is logged in")
-                .isEqualTo("admin");
-        dashboardPage = new BasicWorkFlow().goToDashboard();
+    fun before() {
+        assertThat(LoginWorkFlow().signIn("admin", "admin").loggedInAs())
+                .`as`("Admin is logged in")
+                .isEqualTo("admin")
+        dashboardPage = BasicWorkFlow().goToDashboard()
     }
 
     @Trace(summary = "The administrator can clear and regenerate all of the " +
             "search indexes")
-    @Test(timeout = ZanataTestCase.MAX_SHORT_TEST_DURATION)
+    @Test(timeout = ZanataTestCase.MAX_SHORT_TEST_DURATION.toLong())
     @Ignore("Unstable - sometimes the button isn't ready and sometimes " +
             "the index fails to complete.")
-    public void regenerateSearchIndexes() throws Exception {
-        ManageSearchPage manageSearchPage = dashboardPage
+    @Throws(Exception::class)
+    fun regenerateSearchIndexes() {
+        var manageSearchPage = dashboardPage!!
```

Comment:
Instead of using the "not null assertion", it's often better to use a non-nullable type with `lateinit` (eg `private lateinit var dashboardPage: DashboardBasePage`), since it does get set to a non-null value before use, in this case by the JUnit Before method.